ci/docker/wasm32-wasi/Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM rust:1-slim
+
+RUN apt-get update -y && apt-get install -y --no-install-recommends \
+  ca-certificates \
+  curl wget \
+  software-properties-common lsb-release gpg-agent gnupg gnupg1 gnupg2 \
+  make \
+  perl \
+  gcc \
+  xz-utils \
+  libc6-dev
+
+RUN bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)" -- 14
+RUN update-alternatives --install /usr/bin/llvm-config llvm-config /usr/bin/llvm-config-14 14
+RUN update-alternatives --install /usr/bin/clang clang /usr/bin/clang-14 14
+
+RUN curl https://wasmtime.dev/install.sh -sSf | bash
+RUN cargo install cargo-wasi
```

Comment:
I don't think that `cargo wasi` is really buying anything over what's already present in this CI, so can this be removed?

Reply:
Cargo wasi is taking care of setting up the sysroot, test runner, target args and probably a plenty of other things. I wouldn't want to mess with that personally. It would amount to reimplementing cargo-wasi ourselves, I don't see the point IMO.

All of this is necessary so far